runtime/service/src/test/java/org/apache/polaris/service/events/AttributeKeyTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.events;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+
+class AttributeKeyTest {
+
+  @Test
+  void testCreateAndAccessors() {
+    AttributeKey<String> key = AttributeKey.of("test_key", String.class);
+
+    assertThat(key.name()).isEqualTo("test_key");
+    assertThat(key.type()).isEqualTo(String.class);
+  }
+
+  @Test
+  void testTypeCastValidValue() {
+    AttributeKey<String> key = AttributeKey.of("test_key", String.class);
+
+    String result = key.type().cast("hello");
+
+    assertThat(result).isEqualTo("hello");
+  }
+
+  @Test
+  void testTypeCastNullValue() {
+    AttributeKey<String> key = AttributeKey.of("test_key", String.class);
+
+    String result = key.type().cast(null);
+
+    assertThat(result).isNull();
+  }
+
+  @Test
+  void testTypeCastInvalidType() {
```

Comment:
Similar to above, I would move this test into the test for `PolarisEvent` and test that sending an int to an AttributeKey<String> would throw when calling `key.attribute(..., 123)`.